core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/components/search/search.component.ts

```diff
@@ -159,4 +229,53 @@ export class SearchComponent {
             {} as Record<string, unknown>
         );
     }
+
+    /**
+     * Removes a specific filter and performs a new search.
+     *
+     * @param filterType - The type of filter to remove ('language' | 'site' | 'folder')
+     */
+    removeFilter(filterType: 'language' | 'site' | 'folder') {
+        if (filterType === 'language') {
+            this.form.get('systemSearchableFields.languageId')?.setValue(-1);
+        } else {
+            this.form.get('systemSearchableFields.siteOrFolderId')?.setValue('');
+        }
+
+        this.doSearch();
+    }
+
+    /**
+     * Gets a display-friendly label for the language filter.
+     *
+     * @param languageId - The language ID
+     * @returns A formatted label for display
+     */
+    private getLanguageDisplayLabel(languageId: number): string {
+        const languageValue = this.$languageField()?.languageControl?.value;
+
+        return languageValue?.isoCode || `Language Id: ${languageId}`;
+    }
+
+    /**
+     * Gets a display-friendly label for the site/folder filter.
+     *
+     * @param siteOrFolderId - The site or folder ID in format "type:id"
+     * @returns A formatted label for display, truncated to 70 characters with ellipsis if needed
+     */
+    private getSiteDisplayLabel(siteOrFolderId: string): string {
```

Comment:
this method is used for?

Reply:
get the display value required by design, based on the component value.